lib/symboldatabase.cpp

```diff
@@ -1643,62 +1692,64 @@ void SymbolDatabase::createSymbolDatabaseExprIds()
         }
 
         // Assign IDs
+        ExprIdMap exprIdMap;
+        std::map<std::string, nonneg int> baseIds;
         for (Token* tok = const_cast<Token*>(scope->bodyStart); tok != scope->bodyEnd; tok = tok->next()) {
             if (tok->varId() > 0) {
                 tok->exprId(tok->varId());
-            } else if (isExpression(tok)) {
-                exprs[tok->str()].push_back(tok);
-                tok->exprId(id++);
+                setParentExprId(tok, exprIdMap, id);
```

Comment:
You need to do `followReference` to get the id for the variable because references to the same variable have the same exprId(ie with `int& x = a`, then `x` have the same exprId as `a` because they are the same expression).

Reply:
can you clarify this? Example code:

    void foo(int a) {
        int& r = a;
        return r==a;
    }

The debug output using main branch:

    1: void foo ( int a@var1 ) {
    2: int &@expr1073741827 r@var2 =@expr1073741828 a@var1 ;
    3: return r@var2 ==@expr1073741829 a@var1 ;
    4: }

is `r` getting a expression id here that is not shown? or is my example wrong?